tests/main/snap-network-errors/task.yaml

```diff
@@ -0,0 +1,32 @@
+summary: Ensure network errors are handled gracefully
+
+# autopkgtest run only a subset of tests that deals with the integration
+# with the distro
+backends: [-autopkgtest]
+
+# no iptables on core18 yet
+systems: [-ubuntu-core-18-*]
+
+restore: |
+    echo "Remove iptables rules"
+    iptables -D OUTPUT -p udp --dport 53 -j REJECT --reject-with icmp-port-unreachable || true
+    iptables -D OUTPUT -p tcp --dport 53 -j REJECT --reject-with tcp-reset || true
+
+debug: |
+    echo "iptables rules"
+    iptables -L -n -v
+
+execute: |
+    systemctl stop snapd.{socket,service}
+
+    echo "Disabling DNS queries"
+    iptables -I OUTPUT -p udp --dport 53 -j REJECT --reject-with icmp-port-unreachable
```

Comment:
My guess is `-j REJECT` is what's causing `operation not permitted`, have you tried `-j DROP`?

Reply:
Yeah, I tried, that's not it.

Reply:
Allright, after some more testing it seems a different kind of failure could happen do to forced device auth refresh (I saw that consistently when running on 14.04), so I added `snap refresh` to avoid that.